accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,216 @@
+
+### Health-Check-Tool
+
+* Loki/Promtail: These two components are for log aggregation, indexing and querying. Loki acts as the centralized logging system, storing and managing logs, while Promtail is deployed on the Uyuni server to collect logs and forward them to Loki.
+
+* Grafana: Utilized for visualization, with dashboards displaying metrics, logs, and alerts, facilitating an integrated view of system health and performance.
+
+* Uyuni-health-exporter: A custom exporter to gather specific metrics related to Uyuni server health, including database, service status, and resource usage.
+
+* Alerting System: Leveraging Grafana's alerting mechanism, configured to monitor critical metrics and log patterns, triggering alerts based on predefined thresholds.
+
+* LogCLI: A command-line interface tool provided by Grafana Loki, LogCLI enables direct querying of logs stored in Loki from the terminal. It supports running LogQL queries, live tailing of logs, and viewing logs in a flexible output format. This tool is particularly useful for administrators and developers who prefer to work within a CLI environment for quick log searches, debugging, and log analysis without the need for a graphical interface. 
+
+### Saline
+
+Saline is an addition for salt used in SUSE Manager/Uyuni aimed to provide better control and visibility for states deploymend in the large scale environments. It can provide metrics related to salt events and state apply process on the minions.
+
+
+## Configuration and Deployement
+
+* Monitoring Formula Integration: Loki/Promtail and uyuni_health_exporter configurations are managed via Uyuni's monitoring formula, ensuring seamless integration and ease of deployment.
+
+* Containerized Deployment: The Uyuni-Health-Check components, including Loki, Promtail, Grafana, and the Exporter, are deployed as containers within a POD, simplifying scaling and management.
+
+* Comunication with Uyuni server: The approach to establishing communication with the Uyuni server lies in the underlying infrastructure setup. When deployed in a standalone environment, the system leverages Podman's networking capabilities to facilitate connectivity. This involves configuring Podman containers to ensure they can communicate effectively with the Uyuni server, using Podman's built-in networking features such as container-specific network configurations. Alternatively, in environments where Kubernetes is employed, communication with the Uyuni server is managed through Kubernetes networking principles.
+
+* Storage and Persistence: Utilizing persistent storage solutions for logs and metrics data, ensuring data integrity and availability for historical analysis.
+
+* Container Deployment Function: Design a generic container deployment function in Python that abstracts the container runtime interface. This function will initially support Podman but is designed to allow easy extension to other container orchestration platforms like Kubernetes.
```

Comment:
What is the reasoning behind a "standalone" deployment tool instead of integration with `mgradm`?

Reply:
I've added some notes to clarify this - essentially, we want the Uyuni Health Check tool to be able to run in a "disconnected" enviroment (i.a. when using a supportconfig), therefore an "standalone" deployment function would make sense to avoid having a dependency to this Uyuni server componenent.

Reply:
This conversation continues in another discussion thread